python/GafferSceneUI/SceneInspector.py

```diff
@@ -944,6 +938,9 @@ def __menuDefinition( self, widget = None ) :
 
 		m = IECore.MenuDefinition()
 
+		attribute = targets[0].attributes().get( self.__label().getText() )
```

Comment:
My changes haven't made it through security yet, but I wanted to bring up something I've noticed when looking into it. 
Even before I started messing with the code, inspecting the globals wasn't completely working, was it? When looking at the history of a particular entry, *without having a location selected*, the window is just not gonna pop up because the targets' paths are empty, right? The same thing happens for the ShaderInspector, of course.

Reply:
Ooh, yeah, you're right. Think you could look at fixing that?

Reply:
Yeah, for sure. I'm not sure how, though. 

Should the `Section` tell the `SceneInspector` if it needs the `target.path`? If it does we close the window, otherwise we keep it around because it's gonna show globals, setNames or sets? Would you consider the `Section` to be responsible for providing that kind of information?